rds-proxy/main.tf

```diff
@@ -0,0 +1,72 @@
+resource "aws_iam_role" "proxy_secret_access_role" {
+  name               = "${var.project}-${var.environment}_rds_proxy"
+  assume_role_policy = data.aws_iam_policy_document.assume_role.json
+}
+
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["rds.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role_policy" "proxy_secret_access" {
+  name   = "${var.project}-${var.environment}_rds_proxy"
+  role   = aws_iam_role.proxy_secret_access_role.id
+  policy = data.aws_iam_policy_document.proxy_secret_access_policy.json
+}
+
+data "aws_iam_policy_document" "proxy_secret_access_policy" {
+  statement {
+    effect = "Allow"
+
+    resources = var.db_secret_arns
+
+    actions = ["secretsmanager:GetSecretValue"]
+  }
+}
+
+resource "aws_db_proxy" "proxy" {
+  name                   = "${var.project}-${var.environment}-rds-proxy"
+  debug_logging          = false
+  engine_family          = var.engine
+  role_arn               = aws_iam_role.proxy_secret_access_role.arn
+  vpc_security_group_ids = var.security_groups
+  vpc_subnet_ids         = var.subnets
+
+  dynamic "auth" {
+    for_each = var.db_secret_arns
+    content {
+      auth_scheme = "SECRETS"
+      iam_auth    = "DISABLED"
```

Comment:
Why is this disabled here?

Reply:
because it authenticates using secret instead, IAM auth requires TLS to be setup and it's not needed from the current use-case